lib/status/status.go

```diff
@@ -522,6 +522,15 @@ func fromPlanetAgent(ctx context.Context, local bool, servers []storage.Server)
 		nodes = fromSystemStatus(*status)
 	}
 
+	// Collected system status may exclude the status of some nodes due to
+	// network partition. If offline nodes are detected, degrade system status.
+	for _, node := range nodes {
```

Comment:
Hmm, so is the node going to be excluded completely from the list of nodes in such case (as the comment above suggests), or be present but unhealthy?

Reply:
Currently planet relies on serf to identify the members of the gravity cluster. planet will only be able to construct a SystemStatus that contains the NodeStatus for members of the serf cluster. So the SystemStatus that is queried by gravity may exclude the NodeStatus for some members of the gravity cluster.

gravity maintains its own view of the cluster and does not rely on serf. After querying SystemStatus from planet, if a NodeStatus was not collected for a member, an empty NodeStatus will be initialized for the member. So after this point, a NodeStatus should be present for all members of the gravity cluster.

This added code will detect this case when gravity initializes an empty NodeStatus, and degrade the SystemStatus.